components/layout_2020/flow/mod.rs

```diff
@@ -1632,23 +1630,23 @@ impl PlacementState {
                     self.current_margin
                         .adjoin_assign(&fragment_block_margins.end);
                 } else {
-                    self.current_block_direction_position +=
-                        self.current_margin.solve() + fragment_block_size.into();
+                    self.current_block_direction_position += fragment_block_size.into();
                     self.current_margin = fragment_block_margins.end;
                 }
             },
             Fragment::AbsoluteOrFixedPositioned(fragment) => {
                 let offset = LogicalVec2 {
-                    block: self.current_margin.solve() + self.current_block_direction_position,
-                    inline: Length::new(0.),
+                    block: self.current_margin.solve() +
+                        self.current_block_direction_position.into(),
+                    inline: Au::from_px(0),
                 };
-                fragment.borrow_mut().adjust_offsets(offset);
+                fragment.borrow_mut().adjust_offsets(offset.into());
             },
             Fragment::Float(box_fragment) => {
                 let sequential_layout_state = sequential_layout_state
                     .expect("Found float fragment without SequentialLayoutState");
-                let block_offset_from_containing_block_top =
-                    self.current_block_direction_position + self.current_margin.solve();
+                self.current_block_direction_position += self.current_margin.solve().into();
+                let block_offset_from_containing_block_top = self.current_block_direction_position;
```

Comment:
This also looks quite different from the original code. Before this code was not modifying `self.current_block_direction_position`, but now it is. Why make this change?

Reply:
I apologise. I have addressed this and reverted unnecessary changes.